components/heading/heading-stack.js

```diff
@@ -0,0 +1,41 @@
+import { html, LitElement } from 'lit';
+import { ProviderMixin, requestInstance } from '../../mixins/provider-mixin.js';
+
+export class HeadingLevelController {
```

Comment:
I wonder if there are cases where the consumer would need to reset the level. For instance, in a dialog...? Maybe we would want an optional `level` property to override.

Reply:
That's a really good point yeah -- like either a `level` property like you said, or a boolean `reset` property. If those are set, it would just return that value (or `1` for the `reset`) instead of asking upwards. I'll add this in!

Reply:
I've added `override-level` and am now using it in dialogs as an example.